codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/SymbolVisitor.java

```diff
@@ -0,0 +1,518 @@
+                .putProperty(SymbolUtils.GO_UNIVERSE_TYPE, true)
+                .build();
+    }
+
+    @Override
+    public Symbol booleanShape(BooleanShape shape) {
+        return symbolBuilderFor(shape, "bool").definitionFile("./TypeConversion.go").namespace("typeconversion", "")
+                .putProperty(SymbolUtils.GO_UNIVERSE_TYPE, true)
+                .putProperty("native", symbolBuilderFor(shape, "bool").definitionFile("./TypeConversion.go").build())
+                .putProperty("dafny", symbolBuilderFor(shape, "Wrappers_Compile.Option").addDependency("Wrappers_Compile", "").definitionFile("./TypeConversion.go").build())
+                       .putProperty("ToNative", "val.Dtor_value().(bool)")
+                                              .putProperty("ToDafny", "Wrappers_Compile.Companion_Option_.Create_Some_(val)")
+
+                                              .build();
+    }
+
+    @Override
+    public Symbol listShape(ListShape shape) {
+        return createCollectionSymbol(shape);
+    }
+
+    @Override
+    public Symbol setShape(SetShape shape) {
+        // Go doesn't have a set type. Rather than hack together a set using a map,
+        // we instead just create a list and let the service be responsible for
+        // asserting that there are no duplicates.
+        return createCollectionSymbol(shape);
+    }
+
+    private Symbol createCollectionSymbol(CollectionShape shape) {
+        Symbol reference = toSymbol(shape.getMember());
+        // Shape name will be unused for symbols that represent a slice, but in the event it does we set the collection
+        // shape's name to make debugging simpler.
+        return symbolBuilderFor(shape, getDefaultShapeName(shape))
+                .putProperty(SymbolUtils.GO_SLICE, true)
+                .putProperty(SymbolUtils.GO_UNIVERSE_TYPE,
+                        reference.getProperty(SymbolUtils.GO_UNIVERSE_TYPE, Boolean.class).orElse(false))
+                .putProperty(SymbolUtils.GO_ELEMENT_TYPE, reference)
+                .build();
+    }
+
+    @Override
+    public Symbol mapShape(MapShape shape) {
+        Symbol reference = toSymbol(shape.getValue());
+        // Shape name will be unused for symbols that represent a map, but in the event it does we set the map shape's
+        // name to make debugging simpler.
+        return symbolBuilderFor(shape, getDefaultShapeName(shape))
+                .putProperty(SymbolUtils.GO_MAP, true)
+                .putProperty(SymbolUtils.GO_UNIVERSE_TYPE,
+                        reference.getProperty(SymbolUtils.GO_UNIVERSE_TYPE, Boolean.class).orElse(false))
+                .putProperty(SymbolUtils.GO_ELEMENT_TYPE, reference)
+                .build();
+    }
+
+    private Symbol.Builder symbolBuilderFor(Shape shape, String typeName) {
+//        if (pointableIndex.isPointable(shape)) {
+//            return SymbolUtils.createPointableSymbolBuilder(shape, typeName);
+//        }
```

Comment:
Did we comment this out or did Smithy-Go's authors?
If it was us, why?

Reply:
This was to do away with pointers but based on recent discussion, I'll be handling pointers as part of `TypeConversion`.